mobile/src/screens/survey/components/preferences-step/preferences-step.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+import { Button, Input, ScrollView } from '#libs/components/components';
+import { useAppForm, useCallback, useFormController } from '#libs/hooks/hooks';
+import {
+  getSurveyCategories,
+  type SurveyInputDto,
+  surveyInputValidationSchema,
+} from '#packages/survey/survey';
+import {
+  DEFAULT_SURVEY_PAYLOAD,
+  INVALID_ARRAY_INDEX,
+  PREFERENCES_CATEGORIES,
+  SPLICE_COUNT,
+  TEXTAREA_MAX_LENGTH,
+  TEXTAREA_ROWS_COUNT,
+} from '#screens/survey/libs/constants';
+
+import { SurveyCategory } from '../components';
+import { styles } from './styles';
+
+type Properties = {
+  onSubmit: (options: string[]) => void;
+};
+
+const PreferencesStep: React.FC<Properties> = ({ onSubmit }) => {
+  const {
+    control,
+    errors,
+    isValid,
+    handleSubmit: handleValidationSubmit,
+  } = useAppForm<SurveyInputDto>({
+    defaultValues: DEFAULT_SURVEY_PAYLOAD,
+    validationSchema: surveyInputValidationSchema,
+  });
+  const {
+    field: { onChange: onCategoryChange, value: categoriesValue },
+  } = useFormController({
+    name: 'preferences',
+    control,
+  });
+  const hasOther = categoriesValue.includes('Other');
+
+  const handleFieldChange = useCallback(
+    (option: string) => {
+      const index = categoriesValue.indexOf(option);
+
+      if (index === INVALID_ARRAY_INDEX) {
+        onCategoryChange([...categoriesValue, option]);
+
+        return;
+      }
+
+      categoriesValue.splice(index, SPLICE_COUNT);
+      onCategoryChange(categoriesValue);
+    },
+    [categoriesValue, onCategoryChange],
+  );
+
+  const handleSubmit = useCallback(
```

Comment:
```suggestion
  const handleSurveySubmit = useCallback(
```
And do not rename handleSubmit from useAppForm